bin/pycbc_optimize_snr

```diff
@@ -101,6 +107,14 @@ def compute_network_snr(v, *argv):
         argv = argv[0]
     nsnr, _ = compute_network_snr_core(v, *argv)
     return -nsnr
+    
+def compute_network_snr_pso(v, *argv, **kwargs):
+    argv = kwargs['args']
+    nsnr_array=numpy.zeros(len(v), dtype=float)
+    for i in range(len(v)):
+        nsnr, _ = compute_network_snr_core(v[i], *argv)
+        nsnr_array[i] = nsnr
+    return -nsnr_array
```

Comment:
How large do you expect `v` to be? If it is more than a few tens, you may want to replace the loop with a list comprehension.

Reply:
@titodalcanton Maybe I'm missing something here: Why would a list comprehension be more appropriate if v is large? I don't think there's any internal difference between a list comprehension and a explicit loop with append (maybe there is??) .... For very large v (which this is not), I think this old model is more efficient, as you explicitly declare the memory storage first and then assign into it (it also allows for parallelization if needed).

Reply:
@spxiwh I have definitely seen cases where it made a huge difference, albeit on much older Python versions (maybe loops are more efficient nowadays?).

In this case though I buy your argument that the pre-allocation wins!
```
In [6]: def with_loop(): 
   ...:     x = np.zeros(10000000,dtype=float) 
   ...:     for i in range(10000000): 
   ...:         x[i] = i*i 
   ...:                                                                                             

In [7]: def with_lc(): 
   ...:     x = np.array([i*i for i in range(10000000)]) 
   ...:                                                                                             

In [8]: %time with_loop()                                                                           
CPU times: user 1.23 s, sys: 20.6 ms, total: 1.25 s
Wall time: 1.25 s

In [9]: %time with_lc()                                                                             
CPU times: user 1.31 s, sys: 120 ms, total: 1.43 s
Wall time: 1.43 s
```

Reply:
One could also use numpy's `fromiter()` to avoid creating the list:
```
In [18]: def with_fromiter(): 
    ...:     x = np.fromiter((i*i for i in range(10000000)), dtype=float) 
    ...:                                                                                            

In [19]: %time with_fromiter()                                                                      
CPU times: user 1.1 s, sys: 22 ms, total: 1.12 s
Wall time: 1.12 s
```

Reply:
Is it worth changing it back? I think the length of v only depends on the number of particles, so 250 for the default value.

Reply:
I personally think a list comprehension is more readable in this case, but it appears it may cause a tiny slowdown (it is in a time-critical part of the code) so I am tempted to take my original comment back.

Reply:
I just tested the run-time difference and there doesn't seem to be any noticeable difference (running 5 times for each way, not the largest sample size). I agree that the list comprehension is more readable so I will keep it how it currently is.